src/inject/dynamic-theme/index.ts

```diff
@@ -106,16 +109,28 @@ function createStaticStyleOverrides() {
         `}`
     ].join('\n');
     document.head.insertBefore(variableStyle, inlineStyle.nextSibling);
-    setupStylePositionWatcher(variableStyle, 'variables');
+    setupNodePositionWatcher(variableStyle, 'variables');
+
+    const proxyScript = createOrUpdateElement('darkreader--proxy', undefined, 'script');
+    proxyScript.textContent = `// First get the originial 'native code' of the function.
```

Comment:
I suggest you to move the whole function into a separate file. Avoid polluting window scope.
E.g. use something like
```javascript
import {injectProxy} from './stylesheet-proxy';
proxyScript.textContent = `(${goodFunctionName})()`;
```
It will convert the function to a string and execute.